win32/agent.py

```diff
@@ -171,28 +174,37 @@ def _can_restart(self):
 
     def restart(self):
         if not self._can_restart():
-            servicemanager.LogInfoMsg(
-                "{0} reached the limit of restarts ({1} tries during the last {2}s"
-                " (max authorized: {3})). Not restarting..."
-                .format(self._name, len(self._restarts),
-                        self._RESTART_TIMEFRAME, self._max_restarts)
+            servicemanager.LogWarningMsg(
+                u"`{0}` reached the limit of restarts ({1} tries during the last {2}s"
+                u" (max authorized: {3})). Not restarting..."
+                .format(
+                    self._name, len(self._restarts),
+                    self._RESTART_TIMEFRAME, self._max_restarts
+                )
             )
             self._process.is_enabled = False
             return
 
-        self._restarts.append(time.time())
-        # Make a new proc instances because multiprocessing
-        # won't let you call .start() twice on the same instance.
-        if self._process.is_alive():
-            self._process.terminate()
-
-        # Recreate a new process
-        self._process = self._process.__class__(
-            self._process.config, self._process.hostname,
-            **self._process.options
-        )
-
-        self._process.start()
+        try:
+            # Make a new proc instances because multiprocessing
+            # won't let you call .start() twice on the same instance.
+            if self._process.is_alive():
+                self._process.terminate()
+
+            # Recreate a new process
+            self._process = self._process.__class__(
+                self._process.config, self._process.hostname,
+                **self._process.options
+            )
+            self._process.start()
+        except WindowsError as e:
+            servicemanager.LogErrorMsg(
+                u"Fail to restart `{process}`: {err}".format(
+                    process=self._name, err=e
+                )
+            )
+        else:
+            self._restarts.append(time.time())
```

Comment:
Good call!

Additionally, what would you think about increasing the `SERVICE_SLEEP_INTERVAL` to 2 seconds instead of 1? I want to give more time to the processes to restart,  to avoid "illegitimate" disabling.